src/main/java/uk/gov/companieshouse/overseasentitiesapi/utils/PublicPrivateDataCombiner.java

```diff
@@ -151,4 +167,61 @@ public Map<String, Pair<CompanyOfficerApi, ManagingOfficerDataApi>> buildMergedM
 
     return combineMO;
   }
+
+  public String logCollatedData() {
+    var result = new StringBuilder("{");
+
+    result.append("\"Managing Officer Data\": [");
+
+    Iterator<Entry<String, Pair<CompanyOfficerApi, ManagingOfficerDataApi>>> moIterator = this.combineMO.entrySet().iterator();
```

Comment:
for (Entry<> item : this.combineMO) {
}

Reply:
So I originally had a foreach but unfortunately you need Iterator in an instance to check if it's the last one

Else it prints a comma at the end

Reply:
Add the , before you add an item - except for first when none previously added. Instead of adding after if not last. I just think a foreach would be a lot clearer.

Reply:
It just does not look like modern Java to use an iterator directly. A trick is to add the "," and then remove last after loop if endsin. Or use streams, flatMap and limit,